include/deal.II/base/symmetric_tensor.h

```diff
@@ -1667,13 +1694,42 @@ SymmetricTensor<rank,dim,Number>::operator [] (const TableIndices<rank> &indices
 
 
 
+
+namespace internal
+{
+  namespace SymmetricTensor
+  {
+    template <int dim, typename Number>
+    unsigned int
+    entry_to_indices (const dealii::SymmetricTensor<2,dim,Number> &,
```

Comment:
It's just a way to overload the function based on whether we have a rank-2 or rank-4 tensor (the latter is a few lines down).